service/src/main/kotlin/app/cash/backfila/service/scheduler/RunnerSchedulerService.kt

```diff
@@ -37,6 +37,8 @@ class RunnerSchedulerService @Inject constructor(
 
   override fun run() {
     while (running) {
+      runnerExecutorService.submit { "noop task" }.get()
+      // once the noop runs there is space for another runner
       val newRunners = leaseHunter.hunt()
```

Comment:
should we check `if (!running) break` in case we just got shutdown, so we don't poll for another lease?

Reply:
good call